nifi-nar-bundles/nifi-elasticsearch-bundle/nifi-elasticsearch-restapi-processors/src/test/groovy/org/apache/nifi/processors/elasticsearch/AbstractPutElasticsearchTest.groovy

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License") you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.nifi.processors.elasticsearch
+
+import org.apache.nifi.util.TestRunner
+import org.apache.nifi.util.TestRunners
+import org.junit.jupiter.api.Test
+
+import static org.hamcrest.CoreMatchers.hasItem
+import static org.hamcrest.CoreMatchers.not
+import static org.hamcrest.MatcherAssert.assertThat
+
+abstract class AbstractPutElasticsearchTest<P extends AbstractPutElasticsearch> {
```

Comment:
Please convert this to a Java class as we are starting to move in that direction for testing.

Reply:
NIFI-9290 was raised a while ago to covert all of the unit test classes to java. You chose this as a duplicate of NIFI-10437 although that converted the tests for the Client Service, not these tests

Converting just this class on its own doesn't make sense. Converting them all should be done together in a separate PR, ideally by someone with a tool that can handle a lot of the conversion automatically

I'll reopen NIFI-9290 as it's not been done and want a duplicate

Reply:
Ok. If you'd rather do it that way, I won't block a merge for this reason.